canopen/objectdictionary/eds.py

```diff
@@ -120,7 +134,20 @@ def build_variable(eds, section, index, subindex=0):
             pass
     if eds.has_option(section, "DefaultValue"):
         try:
-            var.default = int(eds.get(section, "DefaultValue"), 0)
+            if var.data_type is objectdictionary.VISIBLE_STRING:
```

Comment:
Check if `data_type` is in `DATA_TYPES` instead.

Reply:
Fixed